### PR TITLE
New Parser for IOSXR - ShowBfdSession

### PR DESCRIPTION
**Pull request by medward2**
## Description
New Parser for IOSXR - ShowBfdSession

## Motivation and Context
parser not present for this command.

## Impact (If any)
none

## Screenshots:
unit testing completed.
(sdk) nso@nso-virtual-machine:~/sdk/genieparser/src/genie/libs/parser/iosxr/tests$ python -m unittest discover
...........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
----------------------------------------------------------------------
Ran 619 tests in 5.326s

OK

## Checklist:

- [ x] I have updated the changelog.
- [ ] I have updated the documentation (If applicable).
- [ x] I have added tests to cover my changes (If applicable).
- [ x] All new and existing tests passed.
- [ x] All new code passed compilation.

### CODE REVIEW

File: src/genie/libs/parser/iosxr/tests/ShowBfdSession/cli/equal/golden_output1_expected.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
+expected_output= {
+    "interface": {
+        "Gi0/0/0/15.103": {
```
Comment (tahigash) [inline src/genie/libs/parser/iosxr/tests/ShowBfdSession/cli/equal/golden_output1_expected.py:3]: we usually convert to full interface name by using `convert_intf_name` function. please do the same.
https://github.com/CiscoTestAutomation/genieparser/blob/master/src/genie/libs/parser/utils/common.py#L485

File: src/genie/libs/parser/iosxr/tests/ShowBfdSession/cli/equal/golden_output1_expected.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            "dest_ip_address": {
+                "172.16.100.130": {
+                    "async": "6s",
```
Comment (tahigash) [inline src/genie/libs/parser/iosxr/tests/ShowBfdSession/cli/equal/golden_output1_expected.py:6]: let's convert to msec all the time and have only number.
```suggestion
                    "async_msec": "6000",
```

File: src/genie/libs/parser/iosxr/tests/ShowBfdSession/cli/equal/golden_output1_expected.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                    "async_multiplier": "(2s*3)",
+                    "echo": "450ms",
```
Comment (tahigash) [inline src/genie/libs/parser/iosxr/tests/ShowBfdSession/cli/equal/golden_output1_expected.py:8]: same with above.
```suggestion
                    "echo_msec": "450",
```

**Review by tahigash — CHANGES_REQUESTED** [review]
reviewed schema and left comments

File: src/genie/libs/parser/iosxr/tests/ShowBfdSession/cli/equal/golden_output1_expected.py (hunk 1 above)
Comment (medward2) [reply src/genie/libs/parser/iosxr/tests/ShowBfdSession/cli/equal/golden_output1_expected.py]: fixed using convert_intf_name function.

File: src/genie/libs/parser/iosxr/tests/ShowBfdSession/cli/equal/golden_output1_expected.py (hunk 2 above)
Comment (medward2) [reply src/genie/libs/parser/iosxr/tests/ShowBfdSession/cli/equal/golden_output1_expected.py]: all fields now converted to msec value and include msec in the group name.

File: src/genie/libs/parser/iosxr/tests/ShowBfdSession/cli/equal/golden_output1_expected.py (hunk 3 above)
Comment (medward2) [reply src/genie/libs/parser/iosxr/tests/ShowBfdSession/cli/equal/golden_output1_expected.py]: all fields now converted to msec value and include msec in the group name.

**Comment by medward2** [comment]
Many thanks. All changes now complete.

regards
mark

**Review by tahigash — APPROVED** [review]
schema looks good now. my teammate will review code!

**Review by choudnik — APPROVED** [review]

**Pull request merged**